go/vt/vttablet/tabletmanager/init_tablet.go

```diff
@@ -155,10 +155,30 @@ func (agent *ActionAgent) InitTablet(port, gRPCPort int32) error {
 	case err == nil:
 		// NOOP
 	case topo.IsErrType(err, topo.NoNode):
-		// try to RebuildKeyspace here but ignore errors if it fails
+		// Try to RebuildKeyspace here but ignore errors if it fails.
+		// It will fail until at least one tablet is up for every shard.
 		topotools.RebuildKeyspace(ctx, logutil.NewConsoleLogger(), agent.TopoServer, *initKeyspace, []string{agent.TabletAlias.Cell})
 	default:
-		return vterrors.Wrap(err, "InitTablet failed to read srvKeyspace")
+		return vterrors.Wrap(err, "InitTablet failed to read SrvKeyspace")
+	}
+
+	// Rebuild vschema graph if this is the first tablet in this keyspace/cell.
+	srvVSchema, err := agent.TopoServer.GetSrvVSchema(ctx, agent.TabletAlias.Cell)
+	switch {
+	case err == nil:
+		// Check if vschema was rebuilt after the initial creation of the keyspace.
+		if _, keyspaceExists := srvVSchema.GetKeyspaces()[*initKeyspace]; !keyspaceExists {
```

Comment:
Doesn't this also need a `RebuildKeyspaceGraph`?

Reply:
I don't understand. Are you saying if a keyspace is missing from SrvVSchema, it implies that RebuildKeyspace has not run for that keyspace?

Reply:
That's what I thought.

Reply:
I thought that SrvKeyspace and SrvVSchema were completely independent. I'll double check.

Reply:
You're probably right. I think they are independent.
But I also think that a missing SrvKeyspace is going to be a problem. And now that I think about it, there are some workflows where this has led to odd errors.
However, I think that's an orthogonal issue, and it's better to resolve that separately.